src/main/java/com/derongan/minecraft/deeperworld/listeners/MovementListener.kt

```diff
@@ -132,44 +162,106 @@ object MovementListener : Listener {
                 }
             }
 
-            player.teleport(newLoc)
-
-            deeperWorld.schedule {
-                waitFor(DeeperConfig.data.entityTeleportDelay)
+            // If the vehicleTree contains a minecart, teleporting the player across worlds within the same tick
+            // as removing the passenger throws a "Removing ticking entity!" exception.
+            // Delay the teleportation by 1 tick in this case.
+            if (vehicleTree.root.where { it.value.type == EntityType.MINECART }.isNotEmpty()) {
```

Comment:
Are there any issues with doing this regardless of whether there's a minecart?

Reply:
Nope, should be fine. I guess we can always wait 1 tick after the passenger removal before teleporting.